passive/find_reflected_params.py

```diff
@@ -0,0 +1,48 @@
+"""
+
+looks for parameter values that are reflected in the response.
+Author: maradrianbelen.com
+
+The scan function will be called for request/response made via ZAP, excluding some of the automated tools
+Passive scan rules should not make any requests 
+
+Note that new passive scripts will initially be disabled
+Right click the script in the Scripts tree and select "enable"
+"""  
+def compare(paramvalue_pair,msg): #get the value in a single parameter and compare it on the HTTP response.
+	value=paramvalue_pair.split('=')
+	#print 'parameter ' + value[0]+ ' is equal to ' + value[1]
+	# add scan for header
+	body=msg.getResponseBody().toString()
+	header=msg.getResponseHeader().toString();
+	reflected=''
+	if body.find(value[1])>-1 or header.find(value[1])>-1:
```

Comment:
This assumes that the parameter has a value, but it might not, leading to errors.